pytest_catchlog.py

```diff
@@ -71,15 +71,22 @@ def catching_logs(handler, filter=None, formatter=None,
                 yield handler
 
 
-def add_option_ini(parser, option, dest, default=None, help=None):
+def add_option_ini(parser, option, dest, action=None, default=None, help=None):
     parser.addini(dest, default=default,
                   help='default value for ' + option)
-    parser.getgroup('catchlog').addoption(option, dest=dest, help=help)
+    parser.getgroup('catchlog').addoption(option,
+                                          action=action,
+                                          dest=dest,
+                                          default=default,
+                                          help=help)
 
-def get_option_ini(config, name):
+
+def get_option_ini(config, name, default=None):
     ret = config.getoption(name)  # 'default' arg won't work as expected
     if ret is None:
         ret = config.getini(name)
+    if ret == '':
+        return default
```

Comment:
But we can't pass an option to the option parser, or else, `get_option_ini` would never get past `ret = config.getoption(name)` because the default value is set.